core/src/data/value/binary_op/u16.rs

```diff
@@ -0,0 +1,1186 @@
+            I16(rhs) => (*rhs as u16) == lhs,
+            I32(rhs) => (*rhs as u16) == lhs,
+            I64(rhs) => (*rhs as u16) == lhs,
+            I128(rhs) => (*rhs as u16) == lhs,
+            U8(rhs) => (*rhs as u16) == lhs,
+            U16(rhs) => *rhs == lhs,
+            F64(rhs) => ((lhs as f64) - rhs).abs() < f64::EPSILON,
+            Decimal(rhs) => Decimal::from(lhs) == *rhs,
+            _ => false,
+        }
+    }
+}
+
+impl PartialOrd<Value> for u16 {
+    fn partial_cmp(&self, other: &Value) -> Option<Ordering> {
+        let lhs = *self;
+        match other {
+            I8(rhs) => lhs.partial_cmp(&(*rhs as u16)),
+            I16(rhs) => lhs.partial_cmp(&(*rhs as u16)),
+            I32(rhs) => lhs.partial_cmp(&(*rhs as u16)),
+            I64(rhs) => lhs.partial_cmp(&(*rhs as u16)),
+            I128(rhs) => lhs.partial_cmp(&(*rhs as u16)),
+            U8(rhs) => self.partial_cmp(&(*rhs as u16)),
+            U16(rhs) => self.partial_cmp(rhs),
+            F64(rhs) => (*self as f64).partial_cmp(rhs),
+            Decimal(rhs) => Decimal::from(*self).partial_cmp(rhs),
+            _ => None,
+        }
+    }
+}
+
+impl TryBinaryOperator for u16 {
+    type Rhs = Value;
+
+    fn try_add(&self, rhs: &Self::Rhs) -> Result<Value> {
+        let lhs = *self;
+
+        match *rhs {
+            I8(rhs) => lhs
+                .checked_add(rhs as u16)
+                .ok_or_else(|| {
+                    ValueError::BinaryOperationOverflow {
+                        lhs: U16(lhs),
+                        rhs: I8(rhs),
+                        operator: NumericBinaryOperator::Add,
+                    }
+                    .into()
+                })
+                .map(U16),
+            I16(rhs) => lhs
+                .checked_add(rhs as u16)
+                .ok_or_else(|| {
+                    ValueError::BinaryOperationOverflow {
+                        lhs: U16(lhs),
+                        rhs: I16(rhs),
+                        operator: NumericBinaryOperator::Add,
+                    }
+                    .into()
+                })
+                .map(U16),
+            I32(rhs) => lhs
+                .checked_add(rhs as u16)
+                .ok_or_else(|| {
+                    ValueError::BinaryOperationOverflow {
+                        lhs: U16(lhs),
+                        rhs: I32(rhs),
+                        operator: NumericBinaryOperator::Add,
+                    }
+                    .into()
+                })
+                .map(U16),
+            I64(rhs) => lhs
+                .checked_add(rhs as u16)
+                .ok_or_else(|| {
+                    ValueError::BinaryOperationOverflow {
+                        lhs: U16(lhs),
+                        rhs: I64(rhs),
+                        operator: NumericBinaryOperator::Add,
+                    }
+                    .into()
+                })
+                .map(U16),
```

Comment:
Looks like we have to compute with large integer like `i64` and convert to u16 at the end like this.

```rs
I64(rhs) => {
    (lhs as i64)
        .checked_add(rhs)
        .ok_or_else(|| {
            ValueError::BinaryOperationOverflow {
                lhs: U16(lhs),
                rhs: I64(rhs),
                operator: NumericBinaryOperator::Add,
            }
            .into()
        })
        .and_then(|v| {
            v.try_into().map_err(|_| {
                ValueError::BinaryOperationOverflow {
                    lhs: U16(lhs),
                    rhs: I64(rhs),
                    operator: NumericBinaryOperator::Add,
                }
                .into()
            })
        })
        .map(U16)
```

*or we may have to fix UnaryOp logic to work after evaluation

Reply:
I think the latter case is better than the former. Since it will make the code much tidier.

Reply:
We decided to throw error as soon as rhs fail while casting.
```rs
lhs.checked_add(rhs.try_into().map_err(|_| {
    ValueError::BinaryOperationOverflow {
        lhs: U16(lhs),
        rhs: I64(rhs),
        operator: NumericBinaryOperator::Add,
    }
})?)
.ok_or_else(|| {
    ValueError::BinaryOperationOverflow {
        lhs: U16(lhs),
        rhs: I64(rhs),
        operator: NumericBinaryOperator::Add,
    }
    .into()
})
.map(U16)
```

For now, It is Okay to repeat error code.
Then, refactor with macro in next PR